OPHD/UI/Core/ToolTip.cpp

```diff
@@ -0,0 +1,80 @@
+#include "ToolTip.h"
+
+#include "../../Cache.h"
+#include "../../Constants.h"
+
+#include <NAS2D/EventHandler.h>
+#include <NAS2D/Utility.h>
+#include <NAS2D/Renderer/Renderer.h>
+
+
+using namespace NAS2D;
+
+
+ToolTip::ToolTip():
+	mFont{ fontCache.load(constants::FONT_PRIMARY, constants::FONT_PRIMARY_NORMAL) }
+{
+	Utility<EventHandler>::get().mouseMotion().connect(this, &ToolTip::onMouseMove);
+}
+
+
+ToolTip::~ToolTip()
+{
+	Utility<EventHandler>::get().mouseMotion().disconnect(this, &ToolTip::onMouseMove);
+}
+
+
+void ToolTip::add(Control& c, const std::string& str)
+{
+	for (auto& item : mControls)
+	{
+		if (item.first == &c)
+		{
+			item.second = str;
+			return;
+		}
+	}

```

Comment:
Much more readable when not using `first` and `second` (nor abbreviations like `c`):
```cpp
	for (auto& item : mControls)
	{
		if (item.control == &targetControl)
		{
			item.toolTipText = str;
			return;
		}
	}
```

Reply:
/me shrugs

I dunno, it's a standard object so to me it's just as readable either way. Though granted, knowing what the pair actually is supposed to be helps. 

For me it's a trivial matter of mousing over a variable and I get this:

![image](https://user-images.githubusercontent.com/13858902/108217932-27690780-7102-11eb-8668-03f69293b1c3.png)

I'm not sure what your environment is like but if you're using something like vim as a text editor instead of a full-fledged IDE with code completion this could be a bit of a drag to work with.

Reply:
I use Atom, which looks like most IDE's, though doesn't have mouse over popups, or any real special C++ code support other than syntax highlighting. The only way I can know what `mControls` is, is to go look for the definition. And there's no jump to definition support, so I actually need to search for it.

Reply:
Which is a valid point. I still think it makes sense to use an IDE that at least allows for this level of code peeking. VS Code may be a better alternative to Atom for this reason.

That stated, your previous answer has me thinking that your suggestion of using `struct`'s over `std::pair` or `std::tuple` is worth taking the time to address.